pkg/operatorsource/configuring_test.go

```diff
@@ -34,19 +35,20 @@ func TestReconcile_NotConfigured_NewCatalogConfigSourceObjectCreated(t *testing.
 		Message: phase.GetMessage(phase.Succeeded),
 	}
 
+	scheme := runtime.NewScheme()
+	marketplace.AddToScheme(scheme)
+
 	datastore := mocks.NewDatastoreWriter(controller)
-	kubeclient := mocks.NewKubeClient(controller)
+	fakeclient := fake.NewFakeClientWithScheme(scheme)
 
-	reconciler := operatorsource.NewConfiguringReconciler(helperGetContextLogger(), datastore, kubeclient)
+	reconciler := operatorsource.NewConfiguringReconciler(helperGetContextLogger(), datastore, fakeclient)
 
 	ctx := context.TODO()
 	opsrcIn := helperNewOperatorSourceWithPhase("marketplace", "foo", phase.Configuring)
 	namespacedName := types.NamespacedName{Name: "opsrc-foo", Namespace: "marketplace"}
 
 	// We expect that the given CatalogConfigSource object does not exist.
 	cscGet := helperNewCatalogSourceConfig(opsrcIn.Namespace, getExpectedCatalogSourceConfigName(opsrcIn.Name))
-	kubeClientErr := k8s_errors.NewNotFound(schema.GroupResource{}, "CatalogSourceConfig not found")
-	kubeclient.EXPECT().Get(context.TODO(), namespacedName, cscGet).Return(kubeClientErr)
```

Comment:
Question: Because the fresh client at this point does not have any record in cache it will return a `NotFound` error for any `get` call. There is no way to ensure that the function being tested passes the expected `namespace`, `name` and `metadata` to the `client.get` function, correct?

Reply:
Correct. I think that the `fakeClient` can only test that the client has the expected state after the function is called.